build/cg/event.rs

```diff
@@ -345,6 +345,28 @@ impl CodeGen {
         }
         writeln!(out, "}}")?;
 
+        writeln!(out)?;
+        writeln!(out, "impl Event {{")?;
+        writeln!(out, "  pub fn as_raw(&self) -> *mut xcb_generic_event_t {{")?;
+        writeln!(out, "    match self {{")?;
+        for event in &self.events {
+            if event.is_xge && self.xcb_mod == "xproto" {
+                // same comment as above
+                continue;
+            }
+            if event.is_xge {
+                // We can't emit the different type, but it is "OK" to cast
+                // it as xcb_generic_event_t as the response_type can be
+                // used to distinguish the correct type
```

Comment:
I agree with the comment. This is how C handles it and we can resolve xge events from `resolve_event` with `xcb_generic_event_t`